cli.js

```diff
@@ -60,8 +60,22 @@ const options = require("commander")
     .option("-f, --macro-file <path>",
         "Read macro definitions, one per line, from the given file.")
     .option("-i, --input <path>", "Read LaTeX input from the given file.")
-    .option("-o, --output <path>", "Write html output to the given file.")
-    .parse(process.argv);
+    .option("-o, --output <path>", "Write html output to the given file.");
+
+// To generate documentation for the CLI, uncomment lines below and
+// run `./cli.js markdown-help`
+/* program.command('markdown-help')
+    .action(() => {
+        console.log(program.options.map(option => `### \`${option.flags}\`
+${option.description}${((option.bool && option.defaultValue !== undefined)
+    ? ' (default: ' + option.defaultValue + ')' : '')}
+`)
+            .concat(['### `-h, --help`\nOutput usage information'])
+            .join('\n'));
+        process.exit();
+    }); */
```

Comment:
Nice idea on how to do this!  Not fully automatic, but close enough.

I'd suggest making this command always available (i.e. uncomment it out), provided that doesn't cause `cli.js` to otherwise break (I assume it runs command-free otherwise, but haven't tested).  And then we can add a comment to the effect "if you change the options or their documentation, please run this command and also update docs/cli.md".

Reply:
@edemaine It outputs the command in the help, i.e., `node cli.js --help`.